sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -65,16 +65,19 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
    */
   override def visitSetConfiguration(ctx: SetConfigurationContext): LogicalPlan = withOrigin(ctx) {
     // Construct the command.
-    val raw = remainder(ctx.SET.getSymbol)
-    val keyValueSeparatorIndex = raw.indexOf('=')
```

Comment:
Okay, so the backquote was added just in case users want to use space in the keys. @maropu, maybe it's a good idea to just reuse `Properties` which we use it when we read Spark configurations. For example:

```scala
import scala.collection.JavaConverters._
val p = new java.util.Properties()
p.load(new java.io.StringReader("a=b"))
p.asScala
```

when there's key only, it will be the empty string as its value. To use whitespaces, we could escape by `\` in `Properties`.

Reply:
Thanks for the comment, @HyukjinKwon. Are you okay with the @cloud-fan idea? https://github.com/apache/spark/pull/29146#issuecomment-663569282

Reply:
Yup, I am okay with that. I thought it'd be easier to use the same codes so we can just say `SET` expects the same syntax as specified in `spark-default.conf`. But using strings look more natural in SQL context. I am okay either way.